src/cli/packer.py

```diff
@@ -0,0 +1,54 @@
+from src.definitions import DiscListContents
+from src.generator import factory as generator_factory
+
+
+class HeadlessGeneratePackWorker():
+    def __init__(self, entry_list: DiscListContents, settings: dict):
+        super().__init__()
+
+        self._generator = generator_factory.get(settings)
+
+        self._entry_list = entry_list
+        self._settings = settings
+        self._progress = 0
+
+    def emit_update_progress(self):
+        self._progress += 1
+
+
+    def generate(self):
```

Comment:
Recommend: Catch IMDException here and sample its status (which will be of type `Status` from `src.definitions`) to print the appropriate helpful message from `StatusMessageDict` so the user isn't guessing what a particular custom exception means. If you want, you could also reraise the IMDException you caught to crash the program afterward so the user gets a stack trace.